packages/cisco_asa/data_stream/log/elasticsearch/ingest_pipeline/default.yml

```
@@ -1266,7 +1266,74 @@ processors:
       trim_key: " "
       trim_value: " "
 
-  #
+  - kv:
+      if: 'ctx._temp_.cisco?.distinguished_name != null && ctx._temp_.cisco.distinguished_name.contains(",")'
+      field: "_temp_.cisco.distinguished_name"
+      tag: "kv_distinguished_name"
+      description: "Split distinguished_name into constituent parts"
+      field_split: ","
+      value_split: "="
+      target_field: "_temp_.cisco.dn_parts"
+      trim_key: " "
+      trim_value: " "
+
+  - script:
+      lang: painless
+      description: >-
+        Sets the parts to the correct names
+      params:
+        "ST":
+          name: "state_or_province"
+        "S":
+          name: "state_or_province"
+        "P":
+          name: "state_or_province"
+        "CN":
+          name: "common_name"
+        "C":
+          name: "country"
+        "L":
+          name: "locality"
+        "O":
+          name: "organization"
+        "OU":
+          name: "organizational_unit"
+      source: |
+          def subject = new HashMap();
+          def parts = ctx._temp_.cisco.dn_parts;
+          for (entry in params.entrySet()) {
+            def k = entry.getKey();
+            def _v = entry.getValue();
+            if (parts.containsKey(k)) {
+              def v = parts[k];
+              List values = (v instanceof List) ? v : [v];   // `[v]` is a Painless list literal
+              subject[_v["name"]] = values;
+            }
+          }
+          ctx._temp_.cisco.dn_parts = subject;
+      if: 'ctx._temp_.cisco?.dn_parts != null'
```

Seems like this can be simplified.

```suggestion
      tag: script_distinguished_name_parts
      description: >-
        Sets the parts to the correct names
      params:
        "ST": "state_or_province"
        "S": "state_or_province"
        "P": "state_or_province"
        "CN": "common_name"
        "C": "country"
        "L": "locality"
        "O": "organization"
        "OU": "organizational_unit"
      source: |
        if (ctx._temp_?.cisco?.dn_parts == null) {
          return;
        }
        def parts = [:];
        ctx._temp_.cisco.dn_parts.forEach((k, v) -> { if (params.containsKey(k)) { parts[params[k]] = v; } else return false; });
        ctx._temp_.cisco.dn_parts = parts;
```

---

yeah, i was doing other things at first. simplifying is good.